Custom/Mails/mailhistoryunit.cpp

```diff
@@ -0,0 +1,82 @@
+#include "mailhistoryunit.h"
+#include "Custom/Mails/ui_mailhistoryunit.h"
+#include "mailhistoryunit.h"
+
+MailHistoryUnit::MailHistoryUnit(QWidget *parent)
+    : QWidget(parent)
+    , ui(new Ui::MailHistoryUnit)
+{
+    ui->setupUi(this);
+    this->installEventFilter(this);
+}
+
+MailHistoryUnit::MailHistoryUnit(const QString& sender, const QString& recipient
+                                 , const QString& subject, const QString& letterBody
+                                 , const QVector<QString> files_paths, QWidget *parent)
+    : MailHistoryUnit(parent)
+{
+    ui->SenderEmailLabel->setText(recipient);
+    ui->LetterSubjectLabel->setText(subject);
+    ui->LetterBodyLabel->setText(letterBody);
```

Comment:
Probably, current lines could be moved to a separate protected or private function, as it repeating quite frequently in another functions